experiments/raja-perf/experiment.py

```diff
@@ -35,29 +36,83 @@ class RajaPerf(
     maintainers("michaelmckinsey1")
 
     def compute_applications_section(self):
+        # TODO: Replace with conflicts clause
+        scaling_modes = {
+            "strong": self.spec.satisfies("+strong"),
+            "weak": self.spec.satisfies("+weak"),
+            "throughput": self.spec.satisfies("+throughput"),
+            "single_node": self.spec.satisfies("+single_node"),
+        }
+
+        scaling_mode_enabled = [key for key, value in scaling_modes.items() if value]
+        if len(scaling_mode_enabled) != 1:
+            raise BenchparkError(
+                f"Only one type of scaling per experiment is allowed for application package {self.name}"
+            )
 
         n_resources = {"n_ranks": 1}
+        problem_sizes = {"size": 1048576}
 
         if self.spec.satisfies("+single_node"):
             for pk, pv in n_resources.items():
                 n_resources = pv
-
+            for nk, nv in problem_sizes.items():
+                self.add_experiment_variable(nk, nv, True)
+        elif self.spec.satisfies("+weak"):
+            # "strong scale" since problem size is per-process
+            scaled_variables = self.generate_strong_scaling_params(
+                {tuple(n_resources.keys()): list(n_resources.values())},
+                int(self.spec.variants["scaling-factor"][0]),
+                int(self.spec.variants["scaling-iterations"][0]),
+            )
+            n_resources = scaled_variables["n_ranks"]
+            for pk, pv in scaled_variables.items():
+                self.add_experiment_variable(pk, pv, True)
+            for nk, nv in problem_sizes.items():
+                self.add_experiment_variable(nk, nv, True)
         elif self.spec.satisfies("+strong"):
             scaled_variables = self.generate_strong_scaling_params(
                 {tuple(n_resources.keys()): list(n_resources.values())},
                 int(self.spec.variants["scaling-factor"][0]),
                 int(self.spec.variants["scaling-iterations"][0]),
             )
             n_resources = scaled_variables["n_ranks"]
+            for pk, pv in scaled_variables.items():
+                self.add_experiment_variable(pk, pv, True)
+            # Notice 1/scaling-factor to keep total problem size constant
+            scaled_problem_sizes = self.generate_strong_scaling_params(
+                {tuple(problem_sizes.keys()): list(problem_sizes.values())},
+                1 / int(self.spec.variants["scaling-factor"][0]),
+                int(self.spec.variants["scaling-iterations"][0]),
+            )
+            problem_sizes = scaled_problem_sizes["size"]
+            for nk, nv in scaled_problem_sizes.items():
+                self.add_experiment_variable(nk, nv, True)
 
-        if self.spec.satisfies("+cuda") or self.spec.satisfies("+rocm"):
+        if self.spec.satisfies("+cuda"):
```

Comment:
Please give RAJAPerf variables specific names, such as RAJAPerf_variant, RAJAPerf_tuning.  Especially since "variant" is a keyword in benchpark/spack.

Reply:
[0e6c92d](https://github.com/LLNL/benchpark/pull/750/commits/0e6c92d1cacb5340e1f63299c09ecd7b537926a9)